Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XRSDK/Editor/WindowsMixedRealityXRSDKConfigurationChecker.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using System.Collections.Generic;
+using System.IO;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.XRSDK.WindowsMixedReality
+{
+    /// <summary>
+    /// Class to perform checks for configuration checks for the WMR XR SDK provider.
+    /// </summary>
+    [InitializeOnLoad]
+    static class WindowsMixedRealityXRSDKConfigurationChecker
+    {
+        private const string AsmDefFileName = "Microsoft.MixedReality.Toolkit.Providers.XRSDK.WMR.asmdef";
+        private const string WindowsMixedRealityReference = "Unity.XR.WindowsMixedReality";
+
+#if UNITY_2019_3_OR_NEWER
+        private static readonly VersionDefine WindowsMixedRealityDefine = new VersionDefine("com.unity.xr.windowsmr", "", "WMR_ENABLED");
+#endif // UNITY_2019_3_OR_NEWER
+
+        static WindowsMixedRealityXRSDKConfigurationChecker()
+        {
+            UpdateAsmDef();
+        }
+
+        /// <summary>
+        /// Updates the assembly definition to contain the appropriate references based on the Unity version.
+        /// </summary>
+        /// <remarks>
+        /// Versions of Unity may have different factorings of components. To address this, the UpdateAsmDef
+        /// method conditionally compiles for the version currently in use.
+        /// To ensure proper compilation on each Unity version, the following steps are performed:
+        /// - Load the Microsoft.MixedReality.Toolkit.Providers.XRSDK.WMR.asmdef file
+        /// - If Unity 2018: nothing
+        /// - If Unity 2019 and newer: Unity.XR.WindowsMixedReality
+        /// - Save the Microsoft.MixedReality.Toolkit.Providers.XRSDK.WMR.asmdef file
+        /// This will result in Unity reloading the assembly with the appropriate dependencies.
+        /// </remarks>
+        private static void UpdateAsmDef()
+        {
+            FileInfo[] asmDefFiles = FileUtilities.FindFilesInAssets(AsmDefFileName);
+
+            if (asmDefFiles.Length == 0)
+            {
+                Debug.LogWarning($"Unable to locate file: {AsmDefFileName}");
+                return;
+            }
+            if (asmDefFiles.Length > 1)
+            {
+                Debug.LogWarning($"Multiple ({asmDefFiles.Length}) {AsmDefFileName} instances found. Modifying only the first.");
+            }
+
+            AssemblyDefinition asmDef = AssemblyDefinition.Load(asmDefFiles[0].FullName);
+            if (asmDef == null)
+            {
+                Debug.LogWarning($"Unable to load file: {AsmDefFileName}");
+                return;
+            }
+
+            List<string> references = new List<string>();
```

Comment:
Super nit because it doesn't actually really matter in small cases like this - I think that references and versionDefines *could* potentially be HashSets (or hashing things) because we do .Contains lookups.

In reality given array of sizes < 10 this should have no noticable effect so take it or leave it (I think in general I prefer using the right data structure for the situation but in these cases they tend to not matter)

Reply:
Ooh thanks, I'll take a look at this!